server/events/working_dir.go

```diff
@@ -232,6 +233,16 @@ func (w *FileWorkspace) forceClone(log logging.SimpleLogging,
 		baseCloneURL = w.TestingOverrideBaseCloneURL
 	}
 
+	var headerCmd []string
+
+	if headRepo.VCSHost.VcsToken != "" {
+		auth := fmt.Sprintf(":%s", headRepo.VCSHost.VcsToken)
+		authBase64Token := base64.StdEncoding.EncodeToString([]byte(auth))
+		headerCmd = []string{"-c", fmt.Sprintf(`http.extraHeader=Authorization: Basic %s`, authBase64Token)}
+	} else {
+		headerCmd = []string{}
+	}
```

Comment:
Can you add a test for this in working_dir_test.go?

Reply:
Looking at this